### PR TITLE
Record opitmizer state

### PR DESCRIPTION
**Pull request by GFabien**
Enable to better resume training by recording the optimizer state when saving the model. Also handle the use of learning rate schedulers from pytorch.

### CODE REVIEW

File: segmentation/config.py [hunk 1]
```diff
@@ -339,9 +340,16 @@ def parse_transform(t):
...
+                if 'prob' in t['transforms'][0]:
+                    t_dict = {
+                        parse_transform(p_and_t['transform']): p_and_t['prob']
+                        for p_and_t in t['transforms']
+                    }
+                else:
+                    t_dict = [
+                        parse_transform(p_and_t['transform']) for p_and_t in
+                        t['transforms']
+                    ]
```
Comment (GFabien) [inline segmentation/config.py:352]: Change this to support not only Crop but also Compose
